### DISCUSSION TITLE
Support for changing Y position of popupBar

### DISCUSSION THREAD

**Discussion by nicholas-t3** [LeoNatan/LNPopupController#429] [General] [answered]
Hello, 

I am an iOS Development student and absolutely love your repo -- I have implemented it in my personal graduation project and it works great. There are several view controllers in my app where I have functions to make the view appear full screen when the user scrolls. However, this creates a bug with my popupBar, as it does not scroll with it, [as seen in this video](https://drive.google.com/file/d/16UJ6nMkBzlyz80wo34uieqfH-otdXgSq/view?usp=sharing)

The normal tab bar is being handled in this function, and called on scrollViewDidScroll: 

`func changeTabBar(hidden:Bool, animated: Bool){
        let tabBar = self.tabBarController?.tabBar
        let offset = (hidden ? UIScreen.main.bounds.size.height : UIScreen.main.bounds.size.height - (tabBar?.frame.size.height)! )
        if offset == tabBar?.frame.origin.y {return}
        let duration:TimeInterval = (animated ? 0.5 : 0.0)
        UIView.animate(withDuration: duration,
                       animations: {tabBar!.frame.origin.y = offset},
                       completion:nil)
    }`

I have tried several ways to get the popupBar to work in a similar function, but do not seem to have enough experience to make it work. When swapping tabBarController.tabBar for .popupBar, I get an error that popupBar doesn't conform to UIView(). I would love some feedback on how to solve this issue so that my project can continue functioning correctly. Thank you!

**Comment by LeoNatan** [comment]
Hello,

As stated here:
https://github.com/LeoNatan/LNPopupController#notes

Direct manipulation of tab bars is not supported (and discouraged by Apple). It’s not a good idea touching the tab bar like that.

**Comment by LeoNatan** [comment, marked as answer]
https://developer.apple.com/documentation/uikit/uitabbarcontroller/1621174-tabbar?language=objc

> You should never attempt to manipulate the UITabBar object itself stored in this property. If you attempt to do so, the tab bar view throws an exception. To configure the items for your tab bar interface, you should instead assign one or more custom view controllers to the viewControllers property. The tab bar collects the needed tab bar items from the view controllers you specify.
> The tab bar view provided by this property is only for situations where you want to display an action sheet using the showFromTabBar: method of the UIActionSheet class.

**Comment by iDevelopper** [comment]
What about trying to dismiss and represent the popup bar with the view controller itself when hiding the tab bar?

**Comment by LeoNatan** [comment]
Won’t work, because the system queries the controller for the tab bar frame. The tab bar controller is not aware of the user frame changes and returns the expected tab bar frame.

**Comment by LeoNatan** [comment]
What might be possible, is overriding the `defaultFrameForBottomDockingView` and trying to provide the user frame there, but that might or might not work.

**Comment by nicholas-t3** [comment]
Thanks for the replies @LeoNatan and @iDevelopper . Sorry for the late reply, this is my first discussion on GitHub and I guess my email notifications aren't on! I was unaware of the Apple documentation in regards to manipulation of the tab bar, and that is good to know, so thank you for bringing that to my attention and pointing it out in your repo notes. 

I will look into the defaultFrameForBottomDockingView method that you are referring to, as I have not heard of that either.  Even though it is discouraged by Apple, I am still interested in finding a solution because it's just a fun graduation project and I think that this unique UI experience stands out. 

Thank you both! I may come back to the discussion with more information but I understand that the pop-up bar is behaving correctly and therefore doesn't warrant any assistance.
